### PR TITLE
feat: handle favorite queries

### PR DESCRIPTION
**Pull request by morganridel**
Based on branch `typeorm`

Related PRs:
 * core: https://github.com/graasp/graasp/pull/379
 * SDK: https://github.com/graasp/graasp-sdk/pull/164 
 * Builder: https://github.com/graasp/graasp-builder/pull/656

### CODE REVIEW

File: src/api/itemFavorite.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import {Item, UUID} from '@graasp/sdk';
+
+import { QueryClientConfig } from '../types';
+import configureAxios, { verifyAuthentication } from './axios';
+import {
+  GET_FAVORITE_ITEMS_ROUTE, buildFavoriteItemRoute
```
Comment (spaenleh) [inline src/api/itemFavorite.ts:6]: Forgot to run prettier ?

File: src/hooks/itemFavorite.test.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { mockHook, setUpTest } from '../../test/utils';
+import {
+  FAVORITE_ITEMS_KEY,
+} from '../config/keys';
+import { GET_FAVORITE_ITEMS_ROUTE } from '../api/routes';
+
+const { hooks, wrapper, queryClient } = setUpTest();
+jest.spyOn(Cookies, 'get').mockReturnValue({ session: 'somesession' });
+
+  describe('useLikesForItem', () => {
```
Comment (spaenleh) [inline src/hooks/itemFavorite.test.ts:14]:
```suggestion
  describe('useFavoriteItems', () => {
```

File: src/hooks/itemFavorite.test.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import { FAVORITE_ITEM, UNAUTHORIZED_RESPONSE } from '../../test/constants';
+import { mockHook, setUpTest } from '../../test/utils';
+import {
+  FAVORITE_ITEMS_KEY,
+} from '../config/keys';
+import { GET_FAVORITE_ITEMS_ROUTE } from '../api/routes';
+
+const { hooks, wrapper, queryClient } = setUpTest();
+jest.spyOn(Cookies, 'get').mockReturnValue({ session: 'somesession' });
```
Comment (spaenleh) [inline src/hooks/itemFavorite.test.ts:12]: I guess we are not using JS cookies anymore

File: src/mutations/itemFavorite.test.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+import { mockMutation, setUpTest, waitForMutation } from '../../test/utils';
+import {
+  buildFavoriteItemRoute,
+} from '../api/routes';
+import { FAVORITE_ITEMS_KEY } from '../config/keys';
+import { addFavoriteItemRoutine, deleteFavoriteItemRoutine } from '../routines';
+import { UNAUTHORIZED_RESPONSE } from '../../test/constants';
+import { StatusCodes } from 'http-status-codes';
+
+jest.spyOn(Cookies, 'get').mockReturnValue({ session: 'somesession' });
```
Comment (spaenleh) [inline src/mutations/itemFavorite.test.ts:17]: Same here, I think we can get rid of this.

File: src/mutations/itemFavorite.test.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    
+          expect(mockedNotifier).toHaveBeenCalledWith(
+            expect.objectContaining({
+              type: addFavoriteItemRoutine.FAILURE,
+            }),
+          );
+        });
+    });
+
+    describe('useDeleteItemLike', () => {
```
Comment (spaenleh) [inline src/mutations/itemFavorite.test.ts:92]:
```suggestion
    describe('useUnfavoriteItem', () => {
```

File: src/mutations/itemFavorite.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    onSettled: () => {
+      queryClient.invalidateQueries(FAVORITE_ITEMS_KEY);
+    },
+  });
+  const useUnfavoriteItem = () =>
+    useMutation<void, unknown, UUID>(MUTATION_KEYS.UNFAVORITE_ITEM);
+
+  return {
+    useFavoriteItem,
+    useUnfavoriteItem
```
Comment (spaenleh) [inline src/mutations/itemFavorite.ts:48]: I am not sure the difference between `useFavoriteItems` and `useFavoriteItem` will be clear. Maybe we could have `useAddFavoriteItem` and `useRemoveFavoriteItem` ? If you have a better idea feel free.

File: test/constants.ts [hunk 7]
```diff
@@ -211,6 +212,13 @@ export const RECYCLED_ITEM_DATA: List<RecycledItemDataRecord> = convertJs([
   },
 ]);
 
+export const FAVORITE_ITEM: ItemFavoriteRecord = convertJs([
+  {
+    id: `favorite-item-id`,
+    item: ITEM_1,
```
Comment (spaenleh) [inline test/constants.ts:218]: missing `createdAt`

**Review by spaenleh — CHANGES_REQUESTED** [review]
🌳 Looks good, just a few comments, really like the simplicity of this since it has been de-coupled 👍

File: src/mutations/itemFavorite.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+import * as Api from '../api';
+import { FAVORITE_ITEMS_KEY, MUTATION_KEYS } from '../config/keys';
+import { addFavoriteItemRoutine, deleteFavoriteItemRoutine } from '../routines';
+import { QueryClientConfig } from '../types';
+
+export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
+  const { notifier } = queryConfig;
+
+  queryClient.setMutationDefaults(MUTATION_KEYS.ADD_FAVORITE_ITEM, {
```
Comment (pyphilia) [inline src/mutations/itemFavorite.ts:13]: Do you think you could refactor this file given this new format? https://github.com/graasp/graasp-query-client/blob/main/src/mutations/itemTag.ts

**Review by pyphilia — APPROVED** [review]
LGTM I've left one comment so we can move forward to a new format for mutations.

**Review by spaenleh — APPROVED** [review]

**Pull request merged**